packages/suite/src/views/wallet/send/Outputs/Address.tsx

```diff
@@ -230,38 +257,52 @@ export const Address = ({ output, outputId, outputsCount }: AddressProps) => {
                             descriptor: address,
                             coin: symbol,
                         };
-                        // 1. If the address is used but unchecksummed, then Suite will automatically
-                        // convert the address to the correct checksummed form and inform the user as described in the OP.
                         const result = await TrezorConnect.getAccountInfo(params);
 
-                        if (result.success) {
-                            const hasHistory = result.payload.history.total !== 0;
-                            if (hasHistory) {
-                                setValue(inputName, toChecksumAddress(address), {
-                                    shouldValidate: true,
-                                });
-                                setHasAddressChecksummed(true);
-
-                                return;
+                        // 1. Validate address checksum.
+                        // Eth addresses are valid without checksum but Trezor displays them as checksummed.
+                        if (!checkAddressCheckSum(address)) {
+                            // 1. If the address is used but unchecksummed, then Suite will automatically
+                            // convert the address to the correct checksummed form and inform the user as described in the OP.
+                            if (result.success) {
+                                const hasHistory = result.payload.history.total !== 0;
+                                if (hasHistory) {
+                                    setValue(inputName, toChecksumAddress(address), {
+                                        shouldValidate: true,
+                                    });
+                                    setHasAddressChecksummed(true);
+
+                                    return;
+                                }
+
+                                // 2. If the address is not checksummed at all and not found in blockbook
+                                // offer to checksum it with a button.
+                                if (!hasHistory && address === address.toLowerCase()) {
+                                    return translationString(
+                                        'TR_ETH_ADDRESS_NOT_USED_NOT_CHECKSUMMED',
+                                    );
+                                }
                             }
-
-                            // 2. If the address is not checksummed at all and not found in blockbook
-                            // offer to checksum it with a button.
-                            if (!hasHistory && address === address.toLowerCase()) {
-                                return translationString('TR_ETH_ADDRESS_NOT_USED_NOT_CHECKSUMMED');
+                        }
+                        //2. Check if address is a contract address
+                        if (!contractAddressWarningDismissed.current && symbol === 'eth') {
```

Comment:
`symbol === 'eth'` limits this to Ethereum mainnet, is this intended or do we want to support other EVMs as well?